### DISCUSSION TITLE
Power-Meter: Timeout

### DISCUSSION THREAD

**Discussion by JanLue4** [syssi/esphome-modbus-solax-x1#52] [Q&A]
Hi!
First of all thank you for the fantastic Project!

I have installed the Solar-Power-Meter yesterday and it works fine. 

My Wifi Tasmota Power Sensor is not sending a new value within 5 Seconds. So the Meter sends a "meter fault".
I have changed the  "power_sensor_inactivity_timeout:" to 60s and it looks good for the first minutes. 
What happens if I delete the entry "power_sensor_inactivity_timeout:"?  Does the Meter still work?

Best Regards 
Jan

**Comment by syssi** [comment]
If you delete the `power_sensor_inactivity_timeout` parameter the default value of 5 seconds will be used again:

https://github.com/syssi/esphome-solax-x1-mini/blob/main/components/solax_meter_gateway/__init__.py#L29-L31

IMO you should multiply the average update interval of your powermeter by 5 and use this value as inactivity timeout. This avoids too much overshooting / exporting power to the grid if the powermeter is unable to provide valid power demands.

**Reply by JanLue4** [reply]
Thank you for the quick answer. 
I tried so get a value every 5 seconds, but it does not work. I will order a Shelly Pro 3 EM to get valid values.

**Reply by syssi** [reply]
How do you read out your smart meter at the moment? What's the update interval of your smart meter in general?

**Comment by JanLue4** [comment]
I used MQTT. The Tasmota Script of the Smart Meter is set at 1s but if I set 5s it doesn't change.

**Reply by syssi** [reply]
Does the Tasmota Script polls the smartmeter or it's a passive read out using an infrared receiver?

**Comment by JanLue4** [comment]
I am using an infrared Receiver, but in the script I can change the update time (if I have understand it correctly)

I am Using this script: https://tasmota.github.io/docs/Smart-Meter-Interface/#efr-sgm-c2c4-sml

**Reply by syssi** [reply]
It looks like your smart meter does publish a measurement per second.

![efr](https://user-images.githubusercontent.com/2735933/226300868-adec628c-8522-4363-8533-c75b9aedb318.png)

You could try to use a ESPHome based approach to improve the performance / throughput:

https://github.com/syssi/esphome-config-examples/blob/main/esp32-sml-energy-meter.yaml

**Reply by JanLue4** [reply]
Thank you. I am an absolutely beginner with ESP Boards, but I will try it.
